cms/models.py

```diff
@@ -153,10 +153,40 @@ def get_context(self, request):
         context["username"] = username
         context["js_settings_json"] = json.dumps(js_settings)
         context["title"] = self.title
+        context["courses_info"] = get_course_run_enrollment_state(self)
 
         return context
 
 
+def get_course_run_enrollment_state(program):
```

Comment:
Since there's no decision being made using information from the program, just the course, can you change this function to accept a course instead of a program? Then you can construct the `state` dict in `get_context` instead